object_database/database_connection.py

```diff
@@ -490,6 +499,12 @@ def _noViewsOutstanding(self):
         with self._lock:
             return not self._versioned_data._version_number_refcount
 
+    def authenticate(self, token):
+        # TODO: do I need to hold self._lock?
```

Comment:
@braxtonmckee some of these classes that have locks, but it is not clear what these locks protect, so I'm not sure when to hold one. This is one such case.